docs/e2e-recipe.md

```diff
@@ -0,0 +1,237 @@
+# End-to-end test recipe
+
+This is a recipe that a contributor can follow to do end-to-end
+testing, using a container registry and GPU-ful Kubernetes cluster
+that the contributor is authorized to use.
+
+## Setup
+
+Start by setting the shell variable `CONTAINER_IMG_REG` to the
+registry that you intend to use. For example, the following might work
+for you.
+
+```shell
+CONTAINER_IMG_REG=quay.io/${LOGNAME}
+```
+
+Configure kubectl to work with the cluster of your choice.
+
+Build and push the requester container image with a command like the
+following. You can omit the `TARGETARCH` if the runtime ISA matches
+your build time ISA.
+
+```shell
+make build-requester CONTAINER_IMG_REG=$CONTAINER_IMG_REG TARGETARCH=amd64
+make push-requester  CONTAINER_IMG_REG=$CONTAINER_IMG_REG
+```
+
+Build the dual-pods controller image. Omit TARGETARCH if not cross-compiling.
+
+```shell
+make build-controller CONTAINER_IMG_REG=$CONTAINER_IMG_REG TARGETARCH=amd64
+```
+
+Instantiate the Helm chart for the dual-pods controller. Specify the
+tag produced by the build above. Specify the name of the ClusterRole
+to use for Node get/list/watch authorization, or omit if not
+needed. NOTE: if you have done this before then you will need to
+delete the old Helm chart instance before re-making it.
+
+```shell
+helm upgrade --install dpctlr charts/dpctlr --set Image="${CONTAINER_IMG_REG}/dual-pods-controller:9010ece" --set NodeViewClusterRole=vcp-node-viewer
+```
+
+## Example 1: cycle server-requesting Pod
+
+Create a ReplicaSet of 1 server-requesting Pod. Following are two
+examples. The first is rather minimal. The second uses caching/staging
+from vcp-cache-manager.
```

Comment:
can you provide a link to the vcp-cache-manager since not everyone knows what it is.

Reply:
It's not public yet, so not sure whether/what to do about this. I will post a link in Slack for now.